include/lsst/afw/table/FieldBase.h

```diff
@@ -211,7 +211,7 @@ struct FieldBase< Array<U> > {
 
     template <typename Derived>
     void setValueDeep(Element * p, ndarray::ExpressionBase<Derived> const & value) const {
-        if (value.template getSize<0>() != _size) {
+        if (value.template getSize<0>() != static_cast<std::size_t>(_size)) {
```

Comment:
It might make more sense for `_size` to be `size_t`, rather than `int`. Or will this cause problems for `ndarray::makeVector`?